### PR TITLE
Add authentication and roles management

### PR DESCRIPTION
**Pull request by AndrewGumenyuk**
- Add interaction with DB
- Add Identity
- Add Simple Injector
- Add AccountController to WebApi for interaction with Identity
- Add WebUi interaction with the WebApi
- Add sidebar item collaborators
- Add logic to API for GET (list of roles)
-Add logic to API for PATCH(change collaborator role)
- Add logic to API for GET (list of collaborators by repo id)
- Add oportunity get collaborators on UI with client auth
- Add API and UI integration
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/8%23discussion_r57219938%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/8%23discussion_r57220105%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/8%23discussion_r57220526%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/8%23discussion_r57222597%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%20bfd33aecf4ab01453e6f435b7f4d03e24f91e7bf%20DevelopGitHub/GitHubExtension/GitHubExtension.IoCManager/App_Start/DependencyConfig.cs%2017%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/8%23discussion_r57220105%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Where%20usages%20this%20class%20in%20project%3F%22%2C%20%22created_at%22%3A%20%222016-03-23T19%3A14%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.IoCManager/App_Start/DependencyConfig.cs%3AL1-53%22%7D%2C%20%22Pull%20bfd33aecf4ab01453e6f435b7f4d03e24f91e7bf%20DevelopGitHub/GitHubExtension/GitHubExtension.Entities/Class1.cs%208%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/8%23discussion_r57219938%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Do%20you%20think%20it%20is%20need%20in%20commit%3F%22%2C%20%22created_at%22%3A%20%222016-03-23T19%3A13%3A41Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%2C%20%7B%22body%22%3A%20%22Sorry%20file%20was%20removed%20from%20solution%20but%20it%20left%20in%20project%20folder.%20We%20are%20fixing%20it%20now%20and%20send%20a%20new%20pull%20request%22%2C%20%22created_at%22%3A%20%222016-03-23T19%3A17%3A23Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12097084%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/AndrewGumenyuk%22%7D%7D%2C%20%7B%22body%22%3A%20%22Please%2C%20don%27t%20horry%20up.%20Because%20I%20see%20you%20have%20a%20looooot%20of%20work.%22%2C%20%22created_at%22%3A%20%222016-03-23T19%3A32%3A06Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Entities/Class1.cs%3AL0-0%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull bfd33aecf4ab01453e6f435b7f4d03e24f91e7bf DevelopGitHub/GitHubExtension/GitHubExtension.Entities/Class1.cs 8'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/8#discussion_r57219938'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Entities/Class1.cs:L0-0</a></b>
- <a href='https://github.com/TerraVenil'></a> Do you think it is need in commit?
- <a href='https://github.com/AndrewGumenyuk'></a> Sorry file was removed from solution but it left in project folder. We are fixing it now and send a new pull request
- <a href='https://github.com/TerraVenil'></a> Please, don't horry up. Because I see you have a looooot of work.
- [ ] <a href='#crh-comment-Pull bfd33aecf4ab01453e6f435b7f4d03e24f91e7bf DevelopGitHub/GitHubExtension/GitHubExtension.IoCManager/App_Start/DependencyConfig.cs 17'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/8#discussion_r57220105'>File: DevelopGitHub/GitHubExtension/GitHubExtension.IoCManager/App_Start/DependencyConfig.cs:L1-53</a></b>
- <a href='https://github.com/TerraVenil'></a> Where usages this class in project?

<a href='https://www.codereviewhub.com/ITsvetkoFF/Kv-013/pull/8?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/ITsvetkoFF/Kv-013/pull/8?approve=1'></a>&nbsp;<a href='https://github.com/ITsvetkoFF/Kv-013/pull/8'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by ITsvetkoFF** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/ITsvetkoFF/Kv-013/pull/8'>Refresh</a>.<a href='#crh-update-none'></a>

File: DevelopGitHub/GitHubExtension/GitHubExtension.Entities/Class1.cs [hunk 1]
```diff
@@ -1,13 +0,0 @@
-﻿
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-
-namespace GitHubExtension.Entities
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Entities/Class1.cs]: Do you think it is need in commit?

File: DevelopGitHub/GitHubExtension/GitHubExtension.IoCManager/App_Start/DependencyConfig.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+using GitHubExtension.Security.WebApi.Library.Services;
+using Microsoft.AspNet.Identity;
+using Microsoft.AspNet.Identity.EntityFramework;
+using SimpleInjector;
+using SimpleInjector.Integration.Web;
+
+namespace GitHubExtension.IoCManager
+{
+    [assembly: PreApplicationStartMethod(typeof(DependencyConfig), "Initialize")]
+    public static class DependencyConfig
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.IoCManager/App_Start/DependencyConfig.cs]: Where usages this class in project?

File: DevelopGitHub/GitHubExtension/GitHubExtension.Entities/Class1.cs (hunk 1 above)
Comment (AndrewGumenyuk) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Entities/Class1.cs]: Sorry file was removed from solution but it left in project folder. We are fixing it now and send a new pull request

Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Entities/Class1.cs]: Please, don't horry up. Because I see you have a looooot of work.

**Pull request closed without merging**